app/models/nic/interface.rb

```diff
@@ -44,6 +44,22 @@ def vlanid
       return ''
     end
 
+    def mtu
+      # Determine a mtu according to the following cascading rules:
+      # 1. if the interface has a v4 subnet with a non-blank mtu, use that
+      # 2. if the interface has a v6 subnet with a non-blank mtu, use that
+      # 3. if no reasonable mtu was determined, then return nil
+      #
+      # In case the v4 and v6 subnet are both present, they should have the same
+      # mtu. If they have a different mtu, this is probably an error in the
+      # user's database. There is no way of determining the real mtu, so we
+      # pick the v4 one unless it turns out to be blank.
+
+      return self.subnet.mtu if self.subnet && self.subnet.mtu.present?
```

Comment:
As the subnet mtu is now a required parameter, we need to change this as the mtu is always present.

How about taking the maximum value from both subnets for now?

```ruby
[self.subnet.mtu, self.subnet6.mtu].select(&:present?).compact.max
```

Reply:
@timogoebel I don't think this a good idea on a network point of view. Maybe the good way to go will be to follow that @ares suggested, adding validation to this. I think we have to enforce this validations: validates that MTU and VLAN ID are identical for subnet and subnet6 on the same nic and hostgroup. This will be mostly ready on my side today, but error messages don't appears out of the box, I have to take a look at this (maybe because select2 is used for subnets in nic and hostgroup forms ?). I will add it as a separate commit so it will be more easy to revert it if you are not OK with this.
I think it will be nice to add filtering on subnet/subnet6 selects to only show subnets compatibles  if one is already selected. If we go this path, I think it will be more easy to add a network layer in the future. As there is no validations at the moment on vlanid, this can break some users, if they don't use the case **D** for their network topology. I can add a rake task to reports if there is some inconsistencies the the database.

Any comment ?

Reply:
> MTU and VLAN ID are identical for subnet and subnet6 on the same nic and hostgroup. 

Makes sense. I like that.

> This will be mostly ready on my side today, but error messages don't appears out of the box, I have to take a look at this (maybe because select2 is used for subnets in nic and hostgroup forms ?)

select2 should not be an issue there I believe. How do you add the errors?





> I think it will be nice to add filtering on subnet/subnet6 selects to only show subnets compatibles if one is already selected

Yes. That would be nice.

> As there is no validations at the moment on vlanid, this can break some users, if they don't use the case D for their network topology.

Yes. The vlan id can be empty right now. To be honest: Digitalocan, AWS, etc don't have the concept of a VLANID. I believe, we need to keep it optional.